workflow/controller/operator.go

```diff
@@ -2854,7 +2854,16 @@ func (woc *wfOperationCtx) getOutboundNodes(nodeID string) []string {
 		}
 		outboundNodes := make([]string, 0)
 		for _, child := range node.Children {
-			outboundNodes = append(outboundNodes, woc.getOutboundNodes(child)...)
+			childNode, err := woc.wf.Status.Nodes.Get(child)
```

Comment:
Why not put all this new logic inside `getOutboundNodes()` - keep that the definitive source of outbound nodes rather than some answers to that question coming from here.

Reply:
hmm, this code change is inside the `getOutboundNodes` function.
There containerType is missing handler for containerset situation causing it to find the wrong node

Reply:
Yes. I forgot this recursed and just assumed it didn't.